contracts/contract/deposit/RocketDepositPool.sol

```diff
@@ -150,28 +164,51 @@ contract RocketDepositPool is RocketBase, RocketDepositPoolInterface, RocketVaul
         uint256 balance = rocketVault.balanceOf("rocketDepositPool");
         uint256 totalEther = 0;
         // Calculate minipool assignments
-        uint256 maxAssignments = rocketDAOProtocolSettingsDeposit.getMaximumDepositAssignments();
-        MinipoolAssignment[] memory assignments = new MinipoolAssignment[](maxAssignments);
-        MinipoolDeposit depositType = MinipoolDeposit.None;
+
         uint256 count = 0;
         uint256 minipoolCapacity = 0;
-        for (uint256 i = 0; i < maxAssignments; ++i) {
-            // Optimised for multiple of the same deposit type
-            if (count == 0) {
```

Comment:
It seems there's no way to handle running out of one deposit type and moving on to the next one?
We only getNextDeposit() on the first loop, so then we're stuck with that deposit type.
If we run out of that queue, I'm not really sure what happens when dequeueMinipoolByDeposit fails because there's nothing more in the queue (the require in that function fails, but I'm not sure where that ends up putting us cuz I'm a solidity newbie).